packages/framework/src/Model/Product/Brand/BrandDetailFriendlyUrlDataProvider.php

```diff
@@ -55,10 +56,13 @@ public function getFriendlyUrlData(DomainConfig $domainConfig): array
 
         $friendlyUrlsData = [];
 
+        $newMethodForDataCreationExists = method_exists($this->friendlyUrlDataFactory, 'createFromIdAndName');
         foreach ($scalarData as $data) {
-            $friendlyUrlData = $this->friendlyUrlDataFactory->create();
-            $friendlyUrlData->name = $data['id'];
-            $friendlyUrlData->id = $data['name'];
+            if ($newMethodForDataCreationExists) {
+                $friendlyUrlData = $this->createFriendlyUrlDataNewWay($data);
```

Comment:
I'm not really sure about the names... Unfortunately I don't have better names...

Reply:
we discussed it - it is not important as the methods will be removed in `8.0` (added @internal annotation to ensure that)